src/it/scala/com/wavesplatform/it/BalancesSpecification.scala

```diff
@@ -0,0 +1,318 @@
+package com.wavesplatform.it
+
+import java.io.IOException
+
+import com.wavesplatform.settings.Constants
+import org.scalatest.concurrent.{IntegrationPatience, ScalaFutures}
+import org.scalatest.{BeforeAndAfterAll, FunSuite, Matchers, RecoverMethods}
+import scorex.transaction.TransactionParser.TransactionType
+
+import scala.concurrent.Await._
+import scala.concurrent.ExecutionContext.Implicits.global
+import scala.concurrent.Future
+import scala.concurrent.duration._
+import scala.util.Random
+
+class BalancesSpecification(allNodes: Seq[Node]) extends FunSuite with Matchers with ScalaFutures
+  with IntegrationPatience with BeforeAndAfterAll with RecoverMethods {
+  val sender = Random.shuffle(allNodes).head
+  val richBalance = sender.address
+
+  var first: String = _
+  var second: String = _
+  var third: String = _
+
+  override def beforeAll(): Unit = {
+    super.beforeAll()
+
+    first = result(sender.createAddress, 5.seconds)
+    second = result(sender.createAddress, 5.seconds)
+    third = result(sender.createAddress, 5.seconds)
+
+    val f = for {
+      fb <- sender.lastBlock
+      txs <- Future.sequence(Seq(
+        sender.transfer(richBalance, first, 200 * Constants.UnitsInWave, sender.fee(TransactionType.TransferTransaction)),
+        sender.transfer(richBalance, second, 100 * Constants.UnitsInWave, sender.fee(TransactionType.TransferTransaction)),
+        sender.transfer(richBalance, third, 100 * Constants.UnitsInWave, sender.fee(TransactionType.TransferTransaction)))
+      )
+      _ <- {
+        val allNodesAllTxsFindInBlockFutures = for {
```

Comment:
The problem with this test is not its asynchronicity, but rather just plain code quality. This snippet is quite simple:

1. retrieve the most recent block from the sender
2. transfer to three recipients
3. wait until all nodes recieve blocks containing transfer transactions
4. make sure all the recipients have the same balance.

This actually should be written as a four-line for-comprehension.

@Tolsi Please rewrite this, extracting all commonly-used expressions into `val`s and making use of `Future.traverse` for parallel queries. This will vastly increase readability.